exp/controllers/azureasomanagedcluster_controller.go

```diff
@@ -153,5 +172,6 @@ func (r *AzureASOManagedClusterReconciler) reconcileDelete(ctx context.Context,
 	defer done()
 	log.V(4).Info("reconciling delete")
```

Comment:
nit: Should this be moved to the top of the function, and another log line be added to remove the finalizer? I was just comparing this to the azuremachine controller https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/controllers/azuremachine_controller.go#L405

Reply:
I don't think that level of consistency exists among all the existing controllers currently. That actually looks like the only place where we log removing the finalizer.